src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
```

Comment:
Here's my diff to double check the scenario:
```diff
diff --git a/src/async-wrap.cc b/src/async-wrap.cc
index a06e163..fa91e1a 100644
--- a/src/async-wrap.cc
+++ b/src/async-wrap.cc
@@ -185,6 +185,8 @@ void AsyncWrap::Initialize(Local<Object> target,
 
 void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
   Environment* env = Environment::from_destroy_ids_idle_handle(handle);
+  v8::SealHandleScope seal(env->isolate());
+  v8::Context::Scope context_scope(env->context());
   // This doesn't leak a handle.
   Local<Function> fn = env->async_hooks_destroy_function();
   int64_t current_id;
```
This never aborts. To double check I added an `Object::New(env->isolate())` right after the `Context::Scope()` which did cause it to abort. So I don't believe this is leaking a handle. But if I am to hoist the `TryCatch` out of the loop it would be necessary to do anyway.

On the note of always using a `SealHandleScope` on the return of an async call. Guess that would only be useful if there was a centralized way to add the wrapper around `uv_run` every time it's about to call the callback. You know if there's a way to wrap wrap all callbacks from `uv_run` in a `SealHandleScope`?